### PR TITLE
Java: Add `BZPOPMIN` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
Copy-paste from https://github.com/aws/glide-for-redis/pull/1255
https://redis.io/docs/latest/commands/bzpopmin/

### CODE REVIEW

**Review by SanHalacogluImproving — APPROVED** [review]

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 1]
```diff
@@ -234,6 +234,31 @@ CompletableFuture<Double> zaddIncr(
      */
     CompletableFuture<Map<String, Double>> zpopmin(String key);
 
+    /**
+     * Blocks the connection until it removes and returns a member with the lowest score from the
+     * sorted sets stored at the specified <code>keys</code>. The sorted sets are checked in the order
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:239]:
```suggestion
     * first non-empty sorted set, with the given <code>keys</code> being checked in the order
```

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 2]
```diff
@@ -1231,6 +1231,37 @@ public void zpopmin(BaseClient client) {
...
+        // nothing popped out - key does not exist
+        assertNull(
+                client
+                        .bzpopmin(new String[] {key3}, REDIS_VERSION.isLowerThan("7.0.0") ? 1. : 0.001)
+                        .get());
+
+        // pops from the second key
+        assertArrayEquals(
+                new Object[] {key2, "c", 2.0}, client.bzpopmin(new String[] {key3, key2}, .5).get());
+
```
Comment (aaron-congo) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1256]: Do we want a test for "key exists but is empty" or do you think it is unnecessary?

Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1661]: It will block the command until timeout. I added this check to another test.

**Comment by Yury-Fridlyand** [comment]
Unblocked + updated.
I updated docs for `BZPOPMAX` and tests according to the last recommendations.

File: java/client/src/test/java/glide/api/models/TransactionTests.java [hunk 3]
```diff
@@ -383,6 +384,13 @@ public void transaction_builds_protobuf_request(BaseTransaction<?> transaction)
         transaction.zpopmin("key", 2);
         results.add(Pair.of(ZPopMin, buildArgs("key", "2")));
 
+        // TODO use buildArgs after merge/rebase on main
```
Comment (SanHalacogluImproving) [inline java/client/src/test/java/glide/api/models/TransactionTests.java:387]: Don't forget to do it :D

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 4]
```diff
@@ -1632,6 +1636,69 @@ public void zpopmin(BaseClient client) {
...
+    @MethodSource("getClients")
+    public void bzpopmin_timeout_check(BaseClient client) {
+        String key = UUID.randomUUID().toString();
+        // create new client with default request timeout (250 millis)
+        try (var testClient =
+                client instanceof RedisClient
+                        ? RedisClient.CreateClient(commonClientConfig().build()).get()
+                        : RedisClusterClient.CreateClient(commonClusterClientConfig().build()).get()) {
+
+            // ensure that commands doesn't time out even if timeout > request timeout
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1691]: Why does it not timeout, is this a special case for blocking commands?

**Review by SanHalacogluImproving — APPROVED** [review]

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 4 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1691]: yes, 0 timeout means no timeout (infinite blocking)

**Pull request merged**